python/pyspark/ml/feature.py

```diff
@@ -968,7 +968,7 @@ class _CountVectorizerParams(JavaParams, HasInputCol, HasOutputCol):
 
     def __init__(self, *args: Any):
         super(_CountVectorizerParams, self).__init__(*args)
-        self._setDefault(minTF=1.0, minDF=1.0, maxDF=2 ** 63 - 1, vocabSize=1 << 18, binary=False)
+        self._setDefault(minTF=1.0, minDF=1.0, maxDF=2**63 - 1, vocabSize=1 << 18, binary=False)
```

Comment:
FWIW, I think this is false positive from Black presumably (per https://peps.python.org/pep-0008/#other-recommendations). I believe we use one space around these operators.

Reply:
Doesn't this fall into this rule?

> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment; however, never use more than one space, and always have the same amount of whitespace on both sides of a binary operator:

```
2**63 - 1
```

The `-` has a lower priority compared to the power operator?

Reply:
Is this style change because we are upgrading `black`?

Reply:
Apparently, yes. I just tested. In the old version it would not touch this change, the new version does. Playing with it actually shows that the behaviour is specifc to the power operator. I'll check if I find some more details in the black releases.

Reply:
Found [the doc](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-breaks-binary-operators) that mentions about this:

> Almost all operators will be surrounded by single spaces, the only exceptions are unary
operators (`+`, `-`, and `~`), and power operators when both operands are simple. For
powers, an operand is considered simple if it's only a NAME, numeric CONSTANT, or
attribute access (chained attribute access is allowed), with or without a preceding
unary operator.

And [the commit](https://github.com/psf/black/commit/6417c99bfdbdc057e4a10aeff9967a751f4f85e9).

Reply:
@ueshin Please see - https://github.com/psf/black/issues/538

Reply:
Looks like `**` is not covered by PEP8, and the main reason of balck change is [consider about readable](https://github.com/psf/black/pull/2095#issuecomment-871694472), so I personaly think black choice is right.